nemo/collections/speechlm2/models/duplex_ear_tts.py

```diff
@@ -0,0 +1,1492 @@
+            continue
+
+        weight_tensors.append(param.float())
+
+    if not weight_tensors:
+        if first_n_layers is not None:
+            logging.info(f"No weights found for first {first_n_layers} layers with prefix '{layer_prefix}'.")
+        else:
+            logging.info("No weights found to rescale in state_dict.")
+        return state_dict
+
+    # Compute global std across selected weights (on CPU)
+    cpu_weights = [p.detach().cpu() for p in weight_tensors]
+    flat = torch.cat([p.flatten() for p in cpu_weights])
+    current_std = float(torch.std(flat))
+    scale = target_std / (current_std + 1e-8)
+
+    logging.info(
+        f"Rescaling state_dict "
+        f"{'(first N layers)' if first_n_layers else '(all layers)'}: "
+        f"current std = {current_std:.6f}, target = {target_std}, scale = {scale:.6f}"
+    )
+
+    # Apply scaling
+    new_state_dict = {}
+    for name, param in state_dict.items():
+        if (
+            torch.is_tensor(param)
+            and param.ndim > 1
+            and (first_n_layers is None or any(name.startswith(pfx) for pfx in prefixes_to_match))
+        ):
+            new_state_dict[name] = param * scale
+        else:
+            new_state_dict[name] = param
+
+    logging.info("Done: weights rescaled.")
+    return new_state_dict
+
+
+class DuplexEARTTS(LightningModule, HFHubMixin):
+    def __init__(self, cfg: dict) -> None:
+        assert isinstance(cfg, dict), (
+            "You must pass the config to DuplexEARTTS as a Python dict to support hyperparameter serialization "
+            f"in PTL checkpoints (we got: '{type(cfg)=}')."
+        )
+        super().__init__()
+        self.save_hyperparameters()
+        # convert dict to config
+        cfg = DictConfig(cfg)
+        self.trainer_config = cfg.get("trainer", None)
+        self.data_cfg = cfg.data
+        self.cfg = cfg.model
+        self.target_sample_rate = cfg.data.target_sample_rate
+        self.source_sample_rate = cfg.data.source_sample_rate
+        self.normalize_text = cfg.data.get("normalize_text", False)
+
+        self.validation_save_path = os.path.join(cfg.exp_manager.explicit_log_dir, "validation_logs")
+
+        # move back text channel by x, in inference it advance the text channel prediction by x frames
+        self.advance_text_channel_by = self.cfg.get("advance_text_channel_by", None)
+
+        # Load ForCausalLM
+        if self.cfg.tts_config.context_hidden_size is not None:
+            self.language_model = self._load_language_model(self.cfg)
+            self.embed_tokens = self._load_embed_tokens(self.cfg)
+            # delete llm because we use it only to get the  embbeding tokens
+            del self.language_model
+
+        # get codec run precision
+        self.audio_codec_run_dtype = getattr(torch, self.cfg.get("audio_codec_run_dtype", "float32"), torch.float32)
+
+        # instanciate eartts model and codec
+        self._load_tts_model(self.cfg)
```

Comment:
- What is the difference between what happens inside of `_load_tts_model()` and `restore_from_pretrained_checkpoint()`? 
- Why do we want to do both of these instead of the model init function?

It seems cleaner if we separate the creation and the loading of a checkpoint. During init() ,we should simply instantiate the classes. After model init, but before training or inference, we should make a call to do checkpoint weight loading.